sdk/attestation/azure-security-attestation/tests/test_attestation_token.py

```diff
@@ -0,0 +1,95 @@
+            attestation_serialized_policy_signing_key1="junk",
+            attestation_serialized_policy_signing_key2="junk",
+            attestation_serialized_isolated_signing_key='yyyy',
+            attestation_isolated_signing_key='xxxx',
+            attestation_isolated_signing_certificate='xxxx',
+            attestation_service_management_url='https://management.core.windows.net/',
+            attestation_location_short_name='xxxx',
+            attestation_client_id='xxxx',
+            attestation_client_secret='secret',
+            attestation_tenant_id='tenant',
+            attestation_isolated_url='https://fakeresource.wus.attest.azure.net',
+            attestation_aad_url='https://fakeresource.wus.attest.azure.net',
+#            attestation_resource_manager_url='https://resourcemanager/zzz'
+        )
+
+class AzureAttestationTokenTest(AzureTestCase):
+
+    def setUp(self):
+        super(AzureAttestationTokenTest, self).setUp()
+
+    @AttestationPreparer()
+    def test_create_unsecured_token(self):
+        token = AttestationToken(body={"val1":[1,2,3]})
+        assert token.get_body() == {"val1":[1,2,3]}
+
+    @AttestationPreparer()
+    def test_create_signing_key(self, attestation_policy_signing_key0, attestation_policy_signing_certificate0):
+        der_cert = base64.b64decode(attestation_policy_signing_certificate0)
+        cert = x509.load_der_x509_certificate(der_cert)
+
+        der_key = base64.b64decode(attestation_policy_signing_key0)
+        key = serialization.load_der_private_key(der_key, password=None)
+
+
+        signer = SigningKey(key, cert)
+        assert signer.certificate.subject == cert.subject
+
+    @AttestationPreparer()
+    def test_create_signer_ecds(self):
+        eckey = ec.generate_private_key(ec.SECP256R1())
+        builder = CertificateBuilder()
+        builder = builder.subject_name(x509.Name([
+            x509.NameAttribute(NameOID.COMMON_NAME, u'attestation.test'),
+        ]))
+        builder = builder.issuer_name(x509.Name([
+            x509.NameAttribute(NameOID.COMMON_NAME, u'attestation.test'),
+        ]))
+
+        one_day = datetime.timedelta(1, 0, 0)
+        builder = builder.not_valid_before(datetime.datetime.today() - one_day)
+        builder = builder.not_valid_after(datetime.datetime.today() + (one_day * 30))
+        builder = builder.serial_number(x509.random_serial_number())        
+        builder = builder.public_key(eckey.public_key())
+        builder = builder.add_extension(x509.SubjectAlternativeName([x509.DNSName(u'attestation.test')]), critical=False)
+        builder = builder.add_extension(x509.BasicConstraints(ca=False, path_length=None), critical=True)
+        certificate = builder.sign(private_key=eckey, algorithm=hashes.SHA256())
+        signer = SigningKey(eckey, certificate)
+        assert  signer.certificate.subject==x509.Name([x509.NameAttribute(NameOID.COMMON_NAME, u'attestation.test')])
```

Comment:
Are these tests making HTTP calls? It looks like they are testing the models and not the clients.

Reply:
If they are not making HTTP calls, they should be split off into a different test class that does not inherit from `AzureTestCase`, reason being is the recording infrastructure acts up when there are tests that do not generate traffic. [Here](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/tables/azure-data-tables/tests/test_table.py#L522-L562) is an example of this split with the `tables` library.

Reply:
Ok, silly question: What should I derive the class from if NOT AzureTestCase. What is the minimal infrastructure I need to be considered a "test case"?

Reply:
You can inherit from `object`, just make sure that the class name starts with `Test` or pytest won't recognize it.